### PR TITLE
Split next() in several methods.

### PR DESCRIPTION
**Pull request by hgaiser**
This PR splits the `next()` method from `Generator` into several methods so that they can be overridden in subclasses. This is necessary to generate targets for "other" (non-regression, non-labels) data.

### CODE REVIEW

File: keras_retinanet/preprocessing/generator.py [hunk 1]
```diff
@@ -149,11 +131,29 @@ def next(self):
...
+        self,
+        image_shape,
+        boxes,
+        num_classes,
+        mask_shape=None,
+        negative_overlap=0.4,
+        positive_overlap=0.5,
+        **kwargs
+    ):
+        return anchor_targets(image_shape, boxes, num_classes, mask_shape, negative_overlap, positive_overlap)
```
Comment (de-vri-es) [inline keras_retinanet/preprocessing/generator.py]: Might rename the free function `anchor_targets` to reflect that it is used for bounding box regression (with labels).

File: keras_retinanet/preprocessing/generator.py [hunk 2]
```diff
@@ -167,4 +167,34 @@ def next(self):
             labels_batch[index, ...]     = labels
             regression_batch[index, ...] = regression
 
-        return image_batch, [regression_batch, labels_batch]
+        return [regression_batch, labels_batch]
+
+    def compute_batches(self, group_index):
```
Comment (de-vri-es) [inline keras_retinanet/preprocessing/generator.py]: The name is slightly misleading. As it is it sounds like it computes the division into batches, but it computes the input and output for a batch.

Some possible alternatives:
 * `compute_batch_input_output`
 * `compute_input_output_batch`
 * `compute_input_output`
 * `compute_batch_blobs`

File: keras_retinanet/preprocessing/generator.py (hunk 1 above)
Comment (hgaiser) [reply keras_retinanet/preprocessing/generator.py]: Done (renamed to `anchor_targets_bbox`).

File: keras_retinanet/preprocessing/generator.py (hunk 2 above)
Comment (hgaiser) [reply keras_retinanet/preprocessing/generator.py]: Renamed to `compute_input_output`.

**Pull request merged**